runner/runners/invoke.go

```diff
@@ -368,71 +369,62 @@ func (inv *Invoker) run(cmd *runner.Command, id runner.RunID, abortCh chan struc
 		rts.execEnd = stamp()
 		rts.outputStart = stamp()
 		if runType == runner.RunTypeScoot {
-			tmp, err := ioutil.TempDir("", "invoke")
-			if err != nil {
-				return runner.FailedStatus(id, errors.NewError(fmt.Errorf("error staging ingestion dir: %v", err), errors.PostExecFailureExitCode),
-					tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-			}
-			uploadTimer := inv.stat.Latency(stats.WorkerUploadLatency_ms).Time()
-			inv.stat.Counter(stats.WorkerUploads).Inc(1)
-			defer func() {
-				os.RemoveAll(tmp)
-				uploadTimer.Stop()
-			}()
-
-			stdoutName := "STDOUT"
-			stderrName := "STDERR"
-			stdlogName := "STDLOG"
-			if err = stageLogFiles(tmp, stdoutName, stderrName, stdlogName, stdout, stderr, stdlog); err != nil {
-				return runner.FailedStatus(id, errors.NewError(err, errors.PostExecFailureExitCode),
-					tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-			}
+			var stderrUrl, stdoutUrl string
+			// only upload logs to a permanent location if a log uploader is initialized
+			if inv.uploader != nil {
+				uploadTimer := inv.stat.Latency(stats.WorkerUploadLatency_ms).Time()
+				defer func() {
+					uploadTimer.Stop()
+				}()
+
+				// generate a unique id that's appended to the job id to create a unique identifier for logs
+				logUid, _ := uuid.NewV4()
+				stdlogName := "stdlog"
+				stderrName := "stderr"
+				stdoutName := "stdout"
+
+				var isAborted bool
+				// upload stdlog
+				logId := fmt.Sprintf("%s_%s/%s", cmd.JobID, logUid, stdlogName)
```

Comment:
Seems like this solidifies the ID of the log in the worker as opposed to leaving it in the hands of the LogUploader. Even if we end up doing something similar in our Uploader it would give us more flexibility there.

Reply:
Creating id here because this ensures that the urls for all stdout/err/log files are similar as sickle just replaces the last "stderr" part with "stdlog". This won't be required once we start passing just stdlog in task status and maybe we can get back to this once we do that. Also, wanted to use the job id in the unique identifier for the log but it does not matter.